src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -146,7 +146,8 @@ public static TokenInfo createNewAccessToken(@Nonnull Main main, @Nonnull String
         }
         AccessTokenInfo accessToken = new AccessTokenInfo(sessionHandle, userId, refreshTokenHash1, expiryTime,
                 parentRefreshTokenHash1, userData, antiCsrfToken, now, lmrt);
-        String token = JWT.createJWT(new Gson().toJsonTree(accessToken), signingKey.privateKey, VERSION.V2);
+
+        String token = JWT.createJWT(Utils.toJsonTreeWithNulls(accessToken), signingKey.privateKey, VERSION.V2);
         return new TokenInfo(token, expiryTime, now);
 
```

Comment:
- Would the JWT recipe also need to change to allow setting for null values similar to how you have changed the session recipe? Please add a test for it as well in the core.
- What about session data (thats stored in the db)

Reply:
- The JWT library we use throws null claims away (https://github.com/auth0/java-jwt/issues/511#issuecomment-914228182)
- In the issue, the user reported that session data works as expected, but I'll extend our tests to cover this.